linker/shared/src/main/scala/org/scalajs/linker/standard/LinkedClass.scala

```diff
@@ -66,7 +66,37 @@ final class LinkedClass(
 
   def fullName: String = Definitions.decodeClassName(encodedName)
 
-  def copy(
+  private[linker] def refined(
+      kind: ClassKind,
+      fields: List[FieldDef],
+      staticMethods: List[Versioned[MethodDef]],
+      memberMethods: List[Versioned[MethodDef]],
+      hasInstances: Boolean,
+      hasInstanceTests: Boolean,
+      hasRuntimeTypeInfo: Boolean
+  ): LinkedClass = {
+    copy(
+        kind = kind,
+        fields = fields,
+        staticMethods = staticMethods,
+        memberMethods = memberMethods,
+        hasInstances = hasInstances,
+        hasInstanceTests = hasInstanceTests,
+        hasRuntimeTypeInfo = hasRuntimeTypeInfo
+    )
+  }
+
+  private[linker] def optimized(
+      staticMethods: List[Versioned[MethodDef]],
+      memberMethods: List[Versioned[MethodDef]]
+  ): LinkedClass = {
+    copy(
+        staticMethods = staticMethods,
+        memberMethods = memberMethods
+    )
+  }
+
+  private def copy(
```

Comment:
I'm OK with this change in principle, but I'm not sure we're on the same page here: We should be allowed to change `LinkingUnit` in a binary incompatible fashion in a minor version. So this is not a major concern (if it were, `ir.Trees` would be in a much worse position).

Reply:
We should be *allowed* to break it in a minor version (not a patch version), but we should *try* not to, as much as possible, in order to break (future) linker plugins.

That's the point of `standard.*` versus the rest: for the rest, we don't care. For `standard`, we try not to break stuff as long as we can avoid it, to preserve the ecosystem of linker plugins.